packages/core/examples/stackProgressive/index.ts

```diff
@@ -0,0 +1,321 @@
+}
+
+async function showStack(
+  stack: string[],
+  viewport,
+  retrieveConfiguration,
+  name: string
+) {
+  cache.purgeCache();
+  imageRetrieveMetadataProvider.clear();
+  if (retrieveConfiguration) {
+    imageRetrieveMetadataProvider.add('stack', retrieveConfiguration);
+  }
+  timingInfo.innerHTML = `<p id="loading" style="margin:0">Loading ${name}</p>`;
+  startTime = Date.now();
+  element.addEventListener(
+    cornerstone.EVENTS.STACK_NEW_IMAGE,
+    newImageFunction
+  );
+  const start = Date.now();
+  // Set the stack on the viewport
+  await viewport.setStack(stack, 0, retrieveConfiguration);
+
+  // Render the image
+  viewport.render();
+  const end = Date.now();
+  const { transferSyntaxUID } = cornerstone.metaData.get(
+    'transferSyntax',
+    stack[0]
+  );
+  document.getElementById('loading').innerText = `Stack render took ${
+    end - start
+  } using ${transferSyntaxUID}`;
+}
+
+/**
+ * Generate the various configurations by using the options on static DICOMweb:
+ * Base lossy/full thumbnail configuration for HTJ2K:
+ * ```
+ * mkdicomweb create -t jhc --recompress true --alternate jhc --alternate-name lossy "/dicom/DE Images for Rad"
+ * ```
+ *
+ * JLS and JLS thumbnails:
+ * ```bash
+ * mkdicomweb create -t jhc --recompress true --alternate jlsLossless --alternate-name jls "/dicom/DE Images for Rad"
+ * mkdicomweb create -t jhc --recompress true --alternate jls --alternate-name jlsThumbnail --alternate-thumbnail "/dicom/DE Images for Rad"
+ * ```
+ *
+ * HTJ2K and HTJ2K thumbnail - lossless:
+ * ```bash
+ * mkdicomweb create -t jhc --recompress true --alternate jhcLossless --alternate-name htj2k "/dicom/DE Images for Rad"
+ * mkdicomweb create -t jhc --recompress true --alternate jhc --alternate-name htj2kThumbnail --alternate-thumbnail "/dicom/DE Images for Rad"
+ * ```
+ */
+const jlsRetrieveOptions = {
+  ...singleRetrieveStages,
+  retrieveOptions: {
+    default: {
+      framesPath: '/jls/',
+    },
+  },
+};
+
+const jlsThumbnailOptions = {
+  ...singleRetrieveStages,
+  retrieveOptions: {
+    default: {
+      framesPath: '/jlsThumbnail/',
+    },
+  },
+};
+
+const jlsMixedOptions = {
+  ...sequentialRetrieveStages,
+  retrieveOptions: {
+    singleFast: {
+      status: ImageQualityStatus.SUBRESOLUTION,
+      framesPath: '/jlsThumbnail/',
+    },
+    singleFinal: {
+      framesPath: '/jls/',
+    },
+  },
+};
+
+const htj2kProgressiveOptions = {
+  ...singleRetrieveStages,
+  retrieveOptions: {
+    single: {
+      streaming: true,
+      streamingDecode: true,
+    },
+  },
+};
+
+const htj2kLossyOptions = {
+  ...sequentialRetrieveStages,
+  retrieveOptions: {
+    singleFast: {
+      status: ImageQualityStatus.LOSSY,
+      framesPath: '/lossy/',
+      streaming: true,
+      streamingDecode: true,
+    },
+  },
+};
+
+const htj2kMixedOptions = {
+  ...sequentialRetrieveStages,
+  retrieveOptions: {
+    singleFinal: {
+      range: 1,
+      streamingDecode: true,
```

Comment:
Do we need the `streamingDecode` here?

Reply:
Maybe we should have two separate one, streamingDecode and rangeDecode for each separate, or `decode` really only